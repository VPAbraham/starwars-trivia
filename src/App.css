.App {
  background: url('./images/starry-night.jpg');
  height: 100vh;
  width: 100vw;
  z-index: -1000;
}

.cockpit {
  background: url('./images/cockpit.png') top center no-repeat;
  background-size: cover;
  height: 100vh;
  width: 100vw;
  z-index: -500;
}