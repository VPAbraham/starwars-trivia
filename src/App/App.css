* {
  box-sizing: border-box;
}

.App {
  background: url('../images/starry-night.jpg');
  height: 100vh;
  width: 100vw;
  z-index: -1000;
}

.cockpit {
  background: url('../images/cockpit.png') top center no-repeat;
  height: 90vh;
  width: 100vw;
  z-index: -500;
}

ul {
  margin: 0;
}

.load-icon {
  color: white;
  font-size: 50px;
  text-align: center;
}