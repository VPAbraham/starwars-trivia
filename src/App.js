import React, { Component } from 'react';
import './App.css';
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";

class App extends Component {
  constructor() {
    super();
    this.state = {

    }
  }
  render() {
    return (
      <div className='App'>
        <h2>App</h2>
      </div>
    );
  }
}

export default App;
