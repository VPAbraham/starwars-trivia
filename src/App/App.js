import React, { Component } from 'react';
import Loading from '../images/SWloadingIcon.gif';
import CharactersData from '../sample-data/characters';
import Favorites from '../Favorites/Favorites'
import Landing from '../Landing/Landing';
import Movies from '../Movies/Movies';
import { getChars, getUrlData } from '../apiCalls'
import './App.css';
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";



class App extends Component {
  constructor() {
    super();
    this.state = {
      error: '',
      isLoading: true,
      movies: [],
      characters: [],
      favorites: [],
      userName: '',
      userQuote: '',
      userSkill: '',
    }
  }

  componentDidMount() {

    fetch('https://swapi.co/api/films/')
      .then(response => response.json())
      .then(data => {
        let dataSubset = []
        const dataTrim = data.results.map(result => {
          const snippet = { title: result.title, episodeId: result.episode_id,
             releaseDate: result.release_date, characters: result.characters, 
             openingCrawl: result.opening_crawl }
          return dataSubset.push(snippet);
        })
        this.setState({
          isLoading: false,
          movies: dataSubset
        }
        )
      })
      .catch(error => this.setState({ error: error.message, isLoading: false }))

  fetch('https://swapi.co/api/films/')
    .then(response => response.json())
      .then(data => {
      this.setState({
        isLoading: true
      })  
      let sortedResults = data.results.sort((a, b) => 
        a.episode_id - b.episode_id);
      
      let castLists = sortedResults.map(data => {
        return data.characters
      });

      let chars = castLists.map(charUrls =>
        getChars(charUrls)
        )
      this.setState({
        characters: chars,
        isLoading: false
      });  
    });

    // fetch('https://swapi.co/api/films/')
    //   .then(response => response.json())
    //   .then(data => console.log('API chain', data))
  }

  
  updateUserInfo = (userInfo) => {
    const { userName, userQuote, skillLevel } = userInfo;
    this.setState({ userName, userQuote, userSkill: skillLevel });
  }

  render() {
    console.log(this.state)
    return (
      <div className='App'>
        <div className='cockpit'>
          <Router>
            <Switch>
              <Route exact path='/'>
                <Landing updateUserInfo={this.updateUserInfo} />
              </Route>
              <Route path='/movies'>
                {this.state.error && <h2>{this.state.error}</h2>}
                <Movies movies={this.state.movies} />
                {this.state.isLoading && 
                <div className='load-icon'>
                  <p>LOADING</p>
                  <img src={Loading} alt='loading' />
                </div>}
              </Route>
              <Route path='/favorites'>
                <Favorites favorites={this.state.favorites}/>
              </Route>
              <Route path='/characters'>
                {!this.state.isLoading && <Characters characters={this.state.characters}/>}
              </Route>
            </Switch>
          </Router>
        </div>
        <footer className='footer footer--user-info'>
          <p className='p p--user-name'>{this.state.userName}</p>
          <p className='p p--user-quote'>{this.state.userQuote}</p>
          <p className='p p--user-skill'>{this.state.userSkill}</p>          
        </footer>  
      </div>
    );
  }
}

export default App;
