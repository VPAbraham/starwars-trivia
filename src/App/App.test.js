import React, { Component } from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update user when updateUserInfo is called', () => {
    const mockUser = { userName: 'Chris', userQuote: 'There is only do', skillLevel: 'Master' };
    wrapper.instance().updateUserInfo(mockUser);
    expect(wrapper.state('userName')).toEqual('Chris');
    expect(wrapper.state('userQuote')).toEqual('There is only do');
    expect(wrapper.state('userSkill')).toEqual('Master');
  });

  it('should clear user info when clearUserInfo is called', () => {
    wrapper.setState({ userName: 'Chris' })
    wrapper.instance().clearUserInfo()
    expect(wrapper.state('userName')).toEqual('');
  })

  it('should add a character to state when addFavCharacter is called', () => {
    const character = { name: 'Yoda', homeworld: 'Yoda Planet', isFavorite: true, relatedFilms: 'yes', species: 'Yoda Species' }
    wrapper.instance().addFavCharacter(character);
    expect(wrapper.state('favorites')).toEqual([character])
  });

  it('should delete characters when removeFavCharacter is called', () => {
    const character = { name: 'Yoda', homeworld: 'Yoda Planet', isFavorite: true, relatedFilms: 'yes', species: 'Yoda Species' }
    wrapper.setState({ favorites: [character] });
    wrapper.instance().removeFavCharacter(character);
    expect(wrapper.state('favorites')).toEqual([])
  })
})