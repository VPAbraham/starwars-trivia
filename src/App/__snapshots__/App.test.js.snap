// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="cockpit"
  >
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <Landing
          updateUserInfo={[Function]}
        />
      </Route>
      <Route
        path="/movies"
      >
        <Movies
          getCurrentMovieChars={[Function]}
          movies={Array []}
        />
        <div
          className="load-icon"
        >
          <p>
            LOADING
          </p>
          <img
            alt="loading"
            src="SWloadingIcon.gif"
          />
        </div>
      </Route>
      <Route
        path="/favorites"
      >
        <Favorites
          favorites={Array []}
          removeFav={[Function]}
        />
      </Route>
      <Route
        path="/characters"
      >
        <Characters
          addFav={[Function]}
          characters={Array []}
        />
      </Route>
    </Switch>
  </div>
  <footer
    className="footer footer--user-info"
  >
    <p
      className="p p--user-name"
    />
    <p
      className="p p--user-quote"
    />
    <p
      className="p p--user-skill"
    />
    <Button
      method={[Function]}
      msg="Eject"
      path="/"
    />
    <Button
      method={null}
      msg="Favs"
      path="/favorites"
    />
  </footer>
</div>
`;
